modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZonesUtil.java

```diff
@@ -160,10 +161,23 @@ static CompoundCondition triggerKeyConditionForZonesChanges(long revision, int z
      * @return Update condition.
      */
     static CompoundCondition triggerScaleUpScaleDownKeysCondition(long scaleUpTriggerRevision, long scaleDownTriggerRevision,  int zoneId) {
-        return and(
-                value(zoneScaleUpChangeTriggerKey(zoneId)).eq(ByteUtils.longToBytes(scaleUpTriggerRevision)),
-                value(zoneScaleDownChangeTriggerKey(zoneId)).eq(ByteUtils.longToBytes(scaleDownTriggerRevision))
-        );
+        SimpleCondition scaleUpCondition;
+
+        if (scaleUpTriggerRevision != 0) {
```

Comment:
Why do you need this? I believe that we can explicitly generate notExists condition if there are no corresponding keys is ms. Meaning, that we don't need 0 and/or empty map as a signal for emptiness.

Reply:
Discussed in person.